client/reader/recent/index.tsx

```diff
@@ -159,93 +157,68 @@ const Recent = ( { viewToggle }: RecentProps ) => {
 		setIsLoading( data?.isRequesting );
 	}, [ data?.isRequesting ] );
 
-	const { data: subscriptionsCount } = SubscriptionManager.useSubscriptionsCountQuery();
-	const hasSubscriptions = subscriptionsCount?.blogs && subscriptionsCount.blogs > 0;
-
 	return (
 		<div className="recent-feed">
-			<div
-				className={ `recent-feed__list-column ${
-					selectedItem && hasSubscriptions ? 'has-overlay' : ''
-				} ${ ! hasSubscriptions ? 'recent-feed--no-subscriptions' : '' }` }
-			>
+			<div className={ `recent-feed__list-column ${ selectedItem ? 'has-overlay' : '' }` }>
 				<div className="recent-feed__list-column-header">
 					<NavigationHeader title={ translate( 'Recent' ) }>{ viewToggle }</NavigationHeader>
 				</div>
 				<div className="recent-feed__list-column-content">
-					{ ! hasSubscriptions ? (
-						<>
-							<p>
-								{ translate(
-									'{{strong}}Welcome!{{/strong}} Follow your favorite sites and their latest posts will appear here. Read, like, and comment in a distraction-free environment. Get started by selecting your interests below:',
-									{
-										components: {
-											strong: <strong />,
-										},
-									}
-								) }
-							</p>
-							<ReaderOnboarding forceShow />
-						</>
-					) : (
-						<DataViews
-							getItemId={ ( item: ReaderPost, index = 0 ) =>
-								item.postId?.toString() ?? `item-${ index }`
-							}
-							view={ view as View }
-							fields={ fields }
-							data={ shownData }
-							onChangeView={ ( newView: View ) =>
-								setView( {
-									type: newView.type,
-									fields: newView.fields ?? [],
-									layout: view.layout,
-									perPage: newView.perPage,
-									page: newView.page,
-									search: newView.search,
-								} )
-							}
-							paginationInfo={ view.search === '' ? defaultPaginationInfo : paginationInfo }
-							defaultLayouts={ { list: {} } }
-							isLoading={ isLoading }
-							selection={ selectedItem ? [ selectedItem.postId?.toString() ] : [] }
-							onChangeSelection={ ( newSelection: string[] ) => {
-								const selectedPost = data?.items?.find(
-									( item: ReaderPost ) => item.postId?.toString() === newSelection[ 0 ]
-								);
-								setSelectedItem( selectedPost || null );
-							} }
-						/>
-					) }
+					<DataViews
```

Comment:
💯 I think pulling all the subscription logic up into `main` makes this much easier to reason about.